### PR TITLE
fix(test): Update the reference Sapling treestate

### PR DESCRIPTION
**Pull request by upbqdn**
## Motivation

Zebra's treestate serialization differs from `zcashd` in the following way:

`zcashd` omits the serialization of empty trailing ommers, while Zebra doesn't. This means that `zcashd` serializes the Sapling treestate for height 419_201 as

019eb30778ddeea84c72e69e07a1689f3c8def3dc0a1939f0edcbe47279069d9310002000150715810d52caf35471d10feb487213fbd95ff209122225b7b65d27a7fb1a44d

Whereas Zebra serializes it as

019eb30778ddeea84c72e69e07a1689f3c8def3dc0a1939f0edcbe47279069d931001f000150715810d52caf35471d10feb487213fbd95ff209122225b7b65d27a7fb1a44d0000000000000000000000000000000000000000000000000000000000

## Specifications & References

The serialized treestate consists of optional, hex-encoded, 32-byte hashes. If the hash is not present, it is serialized as byte 0, i.e., `0x00`. If the hash is present, it is prefixed by byte 1, i.e. `0x01`. The first two hashes in the serialized treestate are the treestate's left and right leaves. These are followed by the serialized length of the vector of ommers. This length is serialized as 1, 3, 5, or 9 bytes. If the length is less than 253, it is serialized as a single byte. The length is then followed by the serialized ommers.

We can now parse the first string, produced by `zcashd`:

- `0119eb30778ddeea84c72e69e07a1689f3c8def3dc0a1939f0edcbe47279069d931` is the serialized left leaf,
- `00` is the serialized right leaf,
- `02` is the serialized length of the vector of ommers,
- `00` is the serialized first ommer,
- `0150715810d52caf35471d10feb487213fbd95ff209122225b7b65d27a7fb1a44d` is the serialized second ommer.

And the second one, produced by Zebra:

- `0119eb30778ddeea84c72e69e07a1689f3c8def3dc0a1939f0edcbe47279069d931` is the serialized left leaf,
- `00` is the serialized right leaf,
- `1f` is the serialized length of the vector of ommers,
- `00` is the serialized first ommer,
- `0150715810d52caf35471d10feb487213fbd95ff209122225b7b65d27a7fb1a44d` is the serialized second ommer
- `0000000000000000000000000000000000000000000000000000000000` are the remaining 29 serialized ommers.

Note that both serializations represent the same treestate.

## Solution

Replace the first string by the second one in the test vector.

### Tests

- None.

### Follow-up Work

- None.

### PR Author's Checklist

- [x] The PR name will make sense to users.
- [ ] The PR provides a CHANGELOG summary.
- [ ] The solution is tested.
- [x] The documentation is up to date.
- [x] The PR has a priority label.

### PR Reviewer's Checklist

- [ ] The PR Author's checklist is complete.
- [ ] The PR resolves the issue.

### CODE REVIEW

File: zebra-test/src/vectors/sapling-treestate-main-0-419-201.txt [hunk 1]
```diff
@@ -1 +1 @@
-019eb30778ddeea84c72e69e07a1689f3c8def3dc0a1939f0edcbe47279069d9310002000150715810d52caf35471d10feb487213fbd95ff209122225b7b65d27a7fb1a44d
\ No newline at end of file
+019eb30778ddeea84c72e69e07a1689f3c8def3dc0a1939f0edcbe47279069d931001f000150715810d52caf35471d10feb487213fbd95ff209122225b7b65d27a7fb1a44d0000000000000000000000000000000000000000000000000000000000
```
Comment (arya2) [inline zebra-test/src/vectors/sapling-treestate-main-0-419-201.txt]: It seems like a newline was added and is causing a test failure.

**Review by arya2 — COMMENTED** [review]
This PR looks good once the extra newline is removed. Thank you for the CI fix!

File: zebra-test/src/vectors/sapling-treestate-main-0-419-201.txt (hunk 1 above)
Comment (upbqdn) [reply zebra-test/src/vectors/sapling-treestate-main-0-419-201.txt]: Thanks for noticing. Fixed in [7c1a617](https://github.com/ZcashFoundation/zebra/pull/9051/commits/7c1a61783318e19c0fa8f68778f4542ae44119a4).

**Review by mpguerra — APPROVED** [review]

**Comment by mpguerra** [comment]
@mergify refresh

**Comment by gustavovalverde** [comment]
@mergify refresh

**Comment by mpguerra** [comment]
@mergify queue

**Pull request merged**